### ISSUE TITLE
[Bug] Keyboard shortcuts in editor for macOS no long work in 1.34.0

### ISSUE THREAD

**Issue by neutron-ice** [WerWolv/ImHex#1757]
### Operating System

MacOS

### What's the issue you encountered?

In the `Pattern Editor` keyboard shortcuts, like CMD+C or CMD+V and some rest no longer work. 

Downgraded to the previous version `1.33.2` – everything is okey.

### How can the issue be reproduced?

When in settings assign new shortcut, e.g. press "CMD+V" instead I see "CTRL+V". When press CTRL+V instead I see "CMD+V". 

But anyway nothing work in editor, irrelevant it is "cmd+v" or "ctrl+v" – work only nowhere else defined Alt+V.
Undo (CMD+Z) work like nowhere else defined "CMD+Backspace/Del".

https://github.com/WerWolv/ImHex/assets/149683770/a80f7093-8712-4619-a305-d99c04daa774

### ImHex Version

1.34.0

### ImHex Build Type

- [ ] Nightly or built from sources

### Installation type

Portable

### Additional context?

Apple M1, macOS 12.6

**Comment by paxcut** [comment]
There seems to be two separate errors reported. One is about the pattern editor where the mac shortcuts no longer work. The pattern editor is undergoing a transition where the shortcuts will be part of the shortcuts manager and will be editable like all the other shortcuts, but only the part that removed the old shortcut code made it to the release. Once the pr that adds the new shortcuts is merged then it should work again.
The other issue is about editing shortcuts in the hex editor and I know nothing about that.

**Comment by sisco0** [comment]
Regarding the `cmd` vs `ctrl` inconvenience,
@neutron-ice 
Could you kindly check if keys are properly interpreted at other applications/websites? e.g. https://en.key-test.ru

**Comment by root670** [comment]
It seems Cmd and Ctrl have been swapped. For example undo should be Cmd-Z but only works with Ctrl-Z now. When pressing cmd or ctrl in the shortcut screen they appear swapped (i.e. press Cmd and it shows up a Ctrl)

**Comment by paxcut** [comment]
There has been a recent [commit](https://discord.com/channels/789833418631675954/789840773281218630/1251883947210702899) that seems to be aimed at this issue but it is missing two important details
1) The pattern editor shortcut code has been moved to be part of the shortcut manager so the changes made in the commit will not be relevant if pr #1700 is merged.
2) The main issue becomes that the shortcuts for Mac are done incorrectly for the entire Imhex and not only for the pattern editor. Searching for changes that affect mac shortcuts point to pr #889 but that code has been in place since Jan 2023 so there must be more recent changes that are causing the the issue as it is reported that 1.33.2 works fine.

**Comment by gkpln3** [comment]
It appears that each time i press CMD it interprets it as CTRL instead, and CTRL is interpreted as CMD.

**Comment by paxcut** [comment]
Does that swapping of keys happen everywhere on Imhex? The pattern editor and the hax editor have independent mappings for keys.In fact if you check the shortcut tab in settings you'll see all the different places where mappings could be different which of course includes the shortcut tab window itself.

**Comment by KevinCCucumber** [comment]
M2 Max Macbook user here. For me, sometimes control is accepted instead of cmd, but not really reproducible.
Using CMD does neither work in the pattern editor nor in the imhex window. Checking in the shortcut settings, control and cmd are indeed swapped for me.

**Comment by zakkhoyt** [comment]
Same issue here (<kbd>cmd</kbd> and <kbd>ctrl</kbd> keys are reversed)/. 

M2 max here. I took a video for you to demo. 

https://github.com/user-attachments/assets/1e222ab8-1235-414c-8ad7-70a863f43355

**Comment by paxcut** [comment]
This  is probably an issue with how imgui handles mac keys because afaik imhex uses imgui to obtain keys in any os. The strange think is that it used to work and now it doesn't, yet there has been no changes to the key handling code in quite some time. I did find older issues in imgui that mention how ctrl and cmd were swapped twice and then one of the swaps was removed. but that should have been fixed by now.

Having re-definable shortcuts means you can still define the shortcuts to be exactly what you prefer. Until we can figure out why keys  are swapped now (and for that we need mac users to debug imhex) Define the shortcuts how you prefer (recently the shortcuts for the pattern editor were added to the list of re-definable shortcuts) and wait for a mac user that can debug imhex to figure it out.

**Comment by MaKiPL** [comment]
Bump, still happens on 1.35.4
Macbook Air M1, MacOS 15.0.1

**Comment by Next-Door-Tech** [comment]
Somehow missed this when searching for existing issues - accidentally created duplicate #1954.

TL;DR: the issue where cmd/ctrl are swapped affects ALL shortcuts (not just the pattern editor), and also affects Intel Macs (I'm on a MBP 16" 2019 w/ a Core i9).

**Comment by Nerwosolek** [comment]
Reporting the same happening on my Mac (MBP 16", Core i7, 2015, Mac OS Monterey 12.7.6).
ImHex 1.35.4.
Cmd swapped with Ctrl.

**Comment by WerWolv** [comment]
@lucasaf04 Everything seems to work as expected for me still. Both on latest master and on the 1.36.2 release
